vmm/src/device_manager.rs

```diff
@@ -3027,14 +3043,33 @@ impl DeviceManager {
                 .map_err(DeviceManagerError::UpdateMemoryForVirtioDevice)?;
         }
 
+        self.update_memory(
+            &Range {
+                iova: new_region.start_addr().raw_value(),
+                size: new_region.len() as u64,
+                user_addr: new_region.as_ptr() as u64,
+            },
+            true,
+        )
+    }
+
+    pub fn update_memory(&self, range: &Range, add: bool) -> DeviceManagerResult<()> {
```

Comment:
Since this is only for VFIO mappings I would prefer it to be named as such.

Reply:
Cool, I agree this will be clearer. The only concern I had was that `vmm/src/vm.rs` would need to be aware of VFIO in a way. But that's okay I'll update!